### PR TITLE
if no wallet name was provided, use one derived from the descriptor

### PR DESCRIPTION
**Pull request by ulrichard**
### Description

if no wallet name was provided, use one derived from the descriptor

### Notes to the reviewers

Is there a better place to use wallet_name_from_descriptor so that the generated name shows up in the unit tests instead of the placeholder?

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk-cli/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Comment by notmandatory** [comment]
How about making `WalletOpts.wallet` an `Option<String>` ? Then you could avoid the "main_marker_will_be_replaced" default value and only do the replacement when `WalletOpts.wallet` is `None`.

**Comment by ulrichard** [comment]
@notmandatory  Yes, of course. Sometimes the solution is so obvious that I have no clue why I didn't find it myself. Thanks.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
tACK https://github.com/bitcoindevkit/bdk-cli/pull/57/commits/773f2dbeb9e272d9cf6ad54d1f8231816a247dd7. Code changes looks good to me.

Below are some minor nits.

I think this warrant an addition in changelog. This is probably a breaking change and we should document this.

File: src/bdk_cli.rs [hunk 1]
```diff
@@ -120,9 +120,11 @@ fn prepare_home_dir() -> PathBuf {
 
 fn open_database(wallet_opts: &WalletOpts) -> Tree {
     let mut database_path = prepare_home_dir();
-    database_path.push(wallet_opts.wallet.clone());
+    let default_wallet_name = "default".to_string();
+    let wallet_name = wallet_opts.wallet.as_ref().unwrap_or(&default_wallet_name);
```
[the remark this answers is not in the document]
Comment (ulrichard) [reply src/bdk_cli.rs:124]: The suggested code leads to: 
> expected `&std::string::String`, found struct `std::string::String`
> help: consider borrowing here: `&"default".to_string()`
And this suggestion leads to:
> creates a temporary which is freed while still in use
That's why I made a separate variable.
But as_deref() did the trick.

File: src/lib.rs [hunk 2]
```diff
@@ -203,18 +203,12 @@ use bdk::{FeeRate, KeychainKind, Wallet};
 ///                   server: "ssl://electrum.blockstream.info:60002".to_string(),
 ///                   stop_gap: 10
 ///               },
-///               #[cfg(feature = "esplora-ureq")]
+///               #[cfg(any(feature = "esplora-ureq", feature = "esplora-reqwest"))]
```
[the remark this answers is not in the document]
Comment (ulrichard) [reply src/lib.rs:206]: Ah yes. I thought about this, was not sure and wanted to try it. But then forgot about it.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
tACK https://github.com/bitcoindevkit/bdk-cli/pull/57/commits/adfdd8c49bdd0683690f55c43abe2ca3c25b5161

Verified that each new descriptor is creating its own wallet directory. Updated checksum mismatch error message is being displayed for wallet with same name but different descriptor.

Just one single nit.

**Review by rajarshimaitra — APPROVED** [review]
ReACK https://github.com/bitcoindevkit/bdk-cli/pull/57/commits/e52e81951e229fd79225349951699a034874b65d

Tested it locally. All seem to be working fine.

There are rooms for few improvement but that can be done in a future PR.

This LGTM.

@notmandatory will wait for you take a look before merging.

**Review by notmandatory — APPROVED** [review]
tACK e52e81951e229fd79225349951699a034874b65d

Works as expected. One suggestion for future PRs is to separate unrelated changes in to different PRs or at least separate commits. In this case the `bdk` version upgrade changes were mixed with the wallet name generation changes which made it a little harder to review. But anyway a good usability improvement!

**Pull request merged**